src/context/WalletProvider/WalletViewsSwitch.tsx

```diff
@@ -35,20 +34,22 @@ export const WalletViewsSwitch = (props: WalletViewProps) => {
   }
 
   useEffect(() => {
-    if (props.routePath) {
-      history.push(props.routePath as string)
+    if (state?.initalRoute) {
+      history.push(state.initalRoute)
     }
-  }, [history, props.routePath])
+  }, [history, state?.initalRoute])
 
   return (
     <>
       <NativePasswordRequired
         onConnect={(wallet: NativeHDWallet) => {
           dispatch({ type: WalletActions.SET_WALLET, payload: wallet })
           dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+          const { name, icon } = SUPPORTED_WALLETS['native']
+          dispatch({ type: WalletActions.SET_WALLET_INFO, payload: { name, icon } })
```

Comment:
When we set a wallet in state should we just automatically update its info inside `SET_WALLET` action?

Reply:
thats a good idea